iac-common/src/test/java/org/sonar/iac/common/extension/IacSensorTest.java

```diff
@@ -355,14 +409,53 @@ protected List<TreeVisitor<InputFileContext>> visitors(SensorContext sensorConte
       protected String getActivationSettingKey() {
         return "testsensor.active";
       }
+    };
+  }
+
+  private IacSensor sensorParseException(CheckFactory checkFactory) {
+    return new IacSensor(SONAR_RUNTIME_8_9, fileLinesContextFactory, noSonarFilter, IacLanguage.IAC) {
+      @Override
+      protected TreeParser<Tree> treeParser() {
+        return testParserThrowsParseException;
+      }
+
+      @Override
+      protected String repositoryKey() {
+        return IacSensorTest.this.repositoryKey();
+      }
+
+      @Override
+      protected List<TreeVisitor<InputFileContext>> visitors(SensorContext sensorContext, DurationStatistics statistics) {
+        return Collections.singletonList(new ChecksVisitor(checkFactory.create(repositoryKey()), statistics));
+      }
+
+      @Override
+      protected String getActivationSettingKey() {
+        return "testsensor.active";
+      }
+    };
+  }
+
+  private IacSensor sensorRecognitionException(CheckFactory checkFactory) {
+    return new IacSensor(SONAR_RUNTIME_8_9, fileLinesContextFactory, noSonarFilter, IacLanguage.IAC) {
+      @Override
+      protected TreeParser<Tree> treeParser() {
+        return testParserThrowsRecognitionException;
+      }
 
       @Override
-      protected ParseException toParseException(String action, InputFile inputFile, Exception cause) {
-        if (!(cause instanceof IOException)) {
-          TextPointer position = new DefaultTextPointer(2,1);
-          return new ParseException("Cannot " + action + " '" + inputFile + "': " + cause.getMessage(), position);
-        }
-        return super.toParseException(action, inputFile, cause);
+      protected String repositoryKey() {
+        return IacSensorTest.this.repositoryKey();
+      }
+
+      @Override
+      protected List<TreeVisitor<InputFileContext>> visitors(SensorContext sensorContext, DurationStatistics statistics) {
+        return Collections.singletonList(new ChecksVisitor(checkFactory.create(repositoryKey()), statistics));
+      }
+
+      @Override
+      protected String getActivationSettingKey() {
+        return "testsensor.active";
```

Comment:
Instead of tripling this method code, I would instead modify the origin sensor method and add a third parameter `TreeParser<Tree> treeParser`
This avoid the code duplication at the code of slightly heavier call to `sensor`.